vehicle/src/main/scala/com/wedriveu/vehicle/simulation/VehicleEventsObservables.scala

```diff
@@ -66,9 +47,10 @@ class VehicleEventsObservablesImpl extends VehicleEventsObservables {
                 return
               }
               //This event will be triggered by a server instruction
-              randomLatitudeDestination = ThreadLocalRandom.current().nextDouble(minorBoundPosition, maxBoundPosition)
-              randomLongitudeDestination = ThreadLocalRandom.current().nextDouble(minorBoundPosition, maxBoundPosition)
-              subscriber.onNext(new Position(randomLatitudeDestination,randomLongitudeDestination ))
+              randomLatitudeDestination = ThreadLocalRandom.current().nextDouble(minorBoundPositionLat, maxBoundPositionLat)
+              randomLongitudeDestination = ThreadLocalRandom.current().nextDouble(minorBoundPositionLon, maxBoundPositionLon)
+              Log.log("Position to reach: " + randomLatitudeDestination + " , " + randomLongitudeDestination)
+              subscriber.onNext(new Position(randomLatitudeDestination, randomLongitudeDestination ))
               Thread.sleep(1500000) //this is temporary
```

Comment:
I know that this is temporary so don't modify it for know, but please remember that we always have to write production-level code, even if we know we will do related task in a short time.
In other words, effort spent in quality is always a good choice! :)